File: java-kata/src/test/java/com/leetcode/ValidParenthesesTest.java
package com.leetcode;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class ValidParenthesesTest {

    private final ValidParentheses it = new ValidParentheses();

    @Test
    void test1() {
//        Input: s = "()"
//        Output: true

        assertThat(it.isValid("()"))
                .isTrue();
    }

    @Test
    void test2() {
//        Input: s = "()[]{}"
//        Output: true

        assertThat(it.isValid("()[]{}"))
                .isTrue();
    }

    @Test
    void test3() {
//        Input: s = "(]"
//        Output: false

        assertThat(it.isValid("(]"))
                .isFalse();
    }

    @Test
    void test4() {
//        Input: s = "([])"
//        Output: true

        assertThat(it.isValid("([])"))
                .isTrue();
    }
}

File: kotlin-kata/src/test/kotlin/com/leetcode/ValidParenthesesTest.kt
package com.leetcode

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class ValidParenthesesTest {

    private val it = ValidParentheses()

    @Test
    fun test1() {
//        Input: s = "()"
//        Output: true

        assertThat(it.isValid("()"))
            .isTrue()
    }

    @Test
    fun test2() {
//        Input: s = "()[]{}"
//        Output: true

        assertThat(it.isValid("()[]{}"))
            .isTrue()
    }

    @Test
    fun test3() {
//        Input: s = "(]"
//        Output: false

        assertThat(it.isValid("(]"))
            .isFalse()
    }

    @Test
    fun test4() {
//        Input: s = "([])"
//        Output: true

        assertThat(it.isValid("([])"))
            .isTrue()
    }

    @Test
    fun test5() {
//        Input: s = "])}"
//        Output: false

        assertThat(it.isValid("])}"))
            .isFalse()
    }
}